bin/strata-client/src/args.rs

```diff
@@ -1,171 +1,230 @@
+#[argh(description = "Strata client")]
 pub struct Args {
-    // TODO: default config location
+    // Config non-overriding args
     #[argh(option, short = 'c', description = "path to configuration")]
-    pub config: Option<PathBuf>,
+    pub config: PathBuf,
 
+    // Config overriding args
+    /// Data directory path that will override the path in the [`Config`].
     #[argh(
         option,
         short = 'd',
         description = "datadir path that will contain databases"
     )]
     pub datadir: Option<PathBuf>,
 
-    #[argh(option, short = 'h', description = "JSON-RPC host")]
-    pub rpc_host: Option<String>,
+    /// Switch that indicates if the client is running as a sequencer.
+    #[argh(switch, description = "is sequencer")]
+    pub sequencer: bool,
 
-    #[argh(option, short = 'r', description = "JSON-RPC port")]
-    pub rpc_port: Option<u16>,
+    /// Rollup params path that will override the params in the [`Config`].
+    #[argh(option, description = "rollup params")]
+    pub rollup_params: Option<PathBuf>,
 
-    #[argh(option, description = "bitcoind RPC host")]
-    pub bitcoind_host: Option<String>,
+    /// Rpc host that the client will listen to.
+    #[argh(option, description = "rpc host")]
+    pub rpc_host: Option<String>,
 
-    #[argh(option, description = "bitcoind RPC user")]
-    pub bitcoind_user: Option<String>,
+    /// Rpc port that the client will listen to.
+    #[argh(option, description = "rpc port")]
+    pub rpc_port: Option<u16>,
 
-    #[argh(option, description = "bitcoind RPC password")]
-    pub bitcoind_password: Option<String>,
+    /// Other generic overrides to the [`Config`].
+    /// Will be used, for example, as `-o btcio.reader.client_poll_dur_ms=1000 -o exec.reth.rpc_url=http://reth`
+    #[argh(option, short = 'o', description = "generic config overrides")]
+    pub overrides: Vec<String>,
+}
 
-    /// Max retries for Bitcoin RPC calls.
-    #[argh(option, description = "max retries for bitcoin RPC (default: 3)")]
-    pub bitcoind_retry_count: Option<u8>,
+impl Args {
+    /// Overrides config. First overrides with the generic overrides passed via `-o` and then
+    /// overrides the result with some of the commonly passed args. Returns if config was overridden
+    /// or not.
+    pub fn override_config(&self, config: &mut Config) -> anyhow::Result<bool> {
+        // Override using -o params.
+        let mut overridden = self.override_generic(config)?;
+
+        // Override by explicitly parsed args like datadir, rpc_host and rpc_port.
+        if let Some(datadir) = &self.datadir {
+            config.client.datadir = datadir.into();
+            overridden = true
+        }
+        if let Some(rpc_host) = &self.rpc_host {
+            config.client.rpc_host = rpc_host.to_string();
+            overridden = true
+        }
+        if let Some(rpc_port) = &self.rpc_port {
+            config.client.rpc_port = *rpc_port;
+            overridden = true
+        }
```

Comment:
These can be lowered to just being overrides instead of having to maintain additional special casing for these.

```rs
fn extract_cmdline_override_strs(args: &Args) -> anyhow::Result<Vec<Override>> {
    let mut overrides = Vec::new();

    if let Some(dd) = &self.datadir {
        overrides.push(Override::new("client.datadir", Value::String(dd.to_owned()));
    }

    // the rest

    Ok(overrides)
}
```

So really you'd have some `.get_overrides()` fn that just returns a list of overrides, the ones explicitly added and then the ones generated from the other options like `-d` and `-p`.  Also I think this is missing a few options.

This pattern also makes it easy when we want to parse values from the env, it's all just the override system.